File: schemas/podcast-platform/drizzle/people.ts
// People who contribute to podcasts as hosts, guests, producers, etc.
// See README.md for full design rationale.

import { pgTable, uuid, text, timestamp, index } from "drizzle-orm/pg-core";
import { files } from "./files";

export const people = pgTable(
  "people",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    slug: text("slug").unique().notNull(),
    bio: text("bio"),
    url: text("url"),
    imageFileId: uuid("image_file_id").references(() => files.id, { onDelete: "set null" }),
    podcastIndexId: text("podcast_index_id"),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_people_name").on(table.name),
  ]
);

File: schemas/podcast-platform/spacetimedb/people.rs
// people: Individuals who contribute to shows and episodes as hosts, guests, or crew.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = people, public)]
pub struct Person {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub name: String,
    #[unique]
    pub slug: String,
    pub bio: Option<String>,
    pub url: Option<String>,
    pub image_file_id: Option<String>, // UUID — FK → files.id (set null on delete)
    pub podcast_index_id: Option<String>,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
